### PR TITLE
[REF] *: refactor old static declaration

### PR DESCRIPTION
**Pull request by bastien-ipb**
Modernization of the codebase to have props/components/whatever properties defined closer to the component
(no need to scroll down to find them, they're now right where we expect
them to be). This is the style that FrameworkJS want to promote for future
developments.

task-3536307

https://github.com/odoo/enterprise/pull/50787

### CODE REVIEW

File: addons/survey/static/src/question_page/question_page_one2many_field.js [hunk 1]
```diff
@@ -25,10 +25,18 @@ function SurveySaveErrorHandler(env, error, originalError) {
     }
 }
 registry
-    .category("error_handlers")
-    .add("surveySaveErrorHandler", SurveySaveErrorHandler, { sequence: 10 });
+.category("error_handlers")
+.add("surveySaveErrorHandler", SurveySaveErrorHandler, { sequence: 10 });
```
Comment (aab-odoo) [inline addons/survey/static/src/question_page/question_page_one2many_field.js:29]: could you remove this diff?

File: addons/web/static/lib/owl/owl.js [hunk 2]
```diff
@@ -2708,6 +2708,7 @@
     }
 
     class Component {
+        static template = "";
```
Comment (aab-odoo) [inline addons/web/static/lib/owl/owl.js:2711]: remove diff in this file. Owl is versioned in its own repository

File: addons/account/static/src/components/account_payment_field/account_payment_field.js [hunk 3]
```diff
@@ -11,14 +11,14 @@ import { formatMonetary } from "@web/views/fields/formatters";
 import { standardFieldProps } from "@web/views/fields/standard_field_props";
 import { Component } from "@odoo/owl";
 
-class AccountPaymentPopOver extends Component {}
-AccountPaymentPopOver.props = {
-    "*": { optional: true },
+class AccountPaymentPopOver extends Component {
+    static props = {"*": { optional: true }}
+    static template = "account.AccountPaymentPopOver"
```
Comment (aab-odoo) [inline addons/account/static/src/components/account_payment_field/account_payment_field.js:16]:
```suggestion
    static props = {
        "*": { optional: true },
    };
    static template = "account.AccountPaymentPopOver";
```

File: addons/loyalty/static/src/js/loyalty_list_view.js [hunk 4]
```diff
@@ -36,13 +37,13 @@ export class LoyaltyActionHelper extends Component {
...
     }
 };
-LoyaltyActionHelper.template = "loyalty.LoyaltyActionHelper";
 
-export class LoyaltyListRenderer extends ListRenderer {};
-LoyaltyListRenderer.template = "loyalty.LoyaltyListRenderer";
-LoyaltyListRenderer.components = {
-    ...LoyaltyListRenderer.components,
-    LoyaltyActionHelper,
+export class LoyaltyListRenderer extends ListRenderer {µ
```
Comment (vava-odoo) [inline addons/loyalty/static/src/js/loyalty_list_view.js:41]: :eyes:

**Review by aab-odoo — COMMENTED** [review]
I didn't read the full diff obviously, but I spotted a few things. Could you fix them and we try to merge this asap as it would be a pain to maintain, and it's not really risky.

Major comment is to remove the diff in owl.js

**Comment by aab-odoo** [comment]
It would be nice to motivate the change a little bit in the commit message. We're doing that to modernize the codebase and to have those props/components/whatever properties defined closer to the component (no need to scroll down to find them, they're now right where we expect them to be). This is the style we want to promote for future developments.

File: addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js [hunk 5]
```diff
@@ -4660,6 +4660,9 @@
 
     class ChartJsComponent extends owl.Component {
         static template = "o-spreadsheet-ChartJsComponent";
+        static props = {
+            figure: Object,
+        };
```
Comment (hokolomopo) [inline addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js:4665]: This file is generated automatically in our repo (https://github.com/odoo/o-spreadsheet), the changes should be there and not it Odoo directly :) 

I can create the PR myself if you don't want to bother going in another repo :)

File: addons/loyalty/static/src/js/loyalty_list_view.js (hunk 4 above)
Comment (pipu-odoo) [reply addons/loyalty/static/src/js/loyalty_list_view.js:41]: :o

File: addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js (hunk 5 above)
Comment (pipu-odoo) [reply addons/spreadsheet/static/src/o_spreadsheet/o_spreadsheet.js:4665]: All right ;)

**Review by aab-odoo — APPROVED** [review]
Very large PR which is hard to maintain, and not really risky, so we r+ it without waiting too long/reviews from all teams, and we'll fix issues/feedbacks later on if there're any.

robodoo r+

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**